### DISCUSSION TITLE
How would I go about deploying the Aspire Dashboard?

### DISCUSSION THREAD

**Discussion by funkysi1701** [microsoft/aspire#931] [Q&A] [answered]
The Aspire Dashboard looks really useful, is this something I can deploy alongside my application or is it just for development purposes to be run from Visual Studio?

**Comment by danmoseley** [comment]
Hi @funkysi1701 it's not Visual Studio dependent at all - it runs on Linux and macOS just the same. 

With respect to extracting it from Aspire, we're supportive and there have been several experiments - maybe @davidfowl  knows the status.

There is also a user story in this repo to enable it for use in deployed apps (in production, in Azure Container Apps for example).

I'd be interested to hear more about your scenario..

**Reply by funkysi1701** [reply]
I've just read @davidfowl comments on this issue https://github.com/dotnet/aspire/issues/932 and this kind of answers my question. Looking forward to watching the progress you make on this project

**Comment by jcvandan** [comment]
I'd also be interested in deploying the dashboard. I just came on to ask the same question. I wouldn't really care what logging system was backing it, would just be nice to have a unified UI for local dev and deployed systems. I am also not a big fan of the app insights UI and prefer a unified dashboard like this.

**Comment by davidfowl** [comment, marked as answer]
We're trying to figure out how to make the dashboard easily deployable. You can do it today but its manual and as is, it won't *just* work as its designed to get data locally only. So there's engineering work we need to do to enable running it outside of local dev environments.

**Reply by 1matin** [reply]
Thanks! And a great hope to see it in the next preview 😀

**Reply by DynConcepts** [reply]
:(  4 months later, and apparently still no....

I am stuck where some things work locally (using main branch) others work only when deployed (not from VS or dotnet cmdline)...having the dashboard would be super useful ...

**Reply by funkysi1701** [reply]
You can use the aspire dashboard docker image, that's what I do. I think preview 4 was the version where this started working well. 

https://learn.microsoft.com/en-us/samples/dotnet/aspire-samples/aspire-standalone-dashboard/

**Reply by plppp2001** [reply]
Can Aspire be deployed to IIS with a core API and Web app?  If so someone please post instructions URL?

**Reply by davidfowl** [reply]
There’s nothing built into aspire for deploying to IIS on Windows servers. You can deploy like how you have always deployed and you can host the dashboard on your Windows server using docker or as a standalone executable (if you install the workload). The dashboard doesn’t run on IIS and there are no plans to support that.

**Comment by cloud9tn** [comment]
Anybody else feel bamboozled here? Microsoft always pushing people to the cloud... even obscurely. It would have been nice to know right up from that this framework is not intended to work locally. I looked at lots of demos and discussions about this, and perhaps I am unlucky, but not a single one expressed this rather massive caveat.

**Reply by ScottGeek** [reply]
It's funny... in my search on what it takes to deploy an "Aspired" app on IIS, this post comes up. Now I fully understand deploying "Containerized" apps on Azure... great wonderful, yes I get it (and I really don't need MS to lecture me on how that was an underlining assumption with Aspire- well, yeah). My thing is that we will forever have on-prem servers (Linux and/or Windows)... they are not going away. Aspire, cool for pro-typing local and development to the cloud.... mostly. But moving to Production with Not-A-Cloud? The surface gets Thin... really fast. The Cloud- Check. Kubernetes- no no no no... IIS/Apache- <cricket noise from MS>.... That is where we are with Aspire.... If you are a Cloud Hosted App developer... go for it. Everyone else... ? Aspire adds no benefit to us that Move to Production (a.k.a. Deploy) with Not-A-Cloud.

**Comment by mip1983** [comment]
I was scratching my head wondering how to deploy the aspire dashboard, it says you can [here](https://ecodriverauthdev.ciamlogin.com/6e75353b-374a-40fa-9f52-37ad7b3c49b3/oauth2/v2.0/authorize?client_id=4f45b1a6-14a2-4b83-b601-269acf74e1f2&response_type=id_token&redirect_uri=https://jwt.ms&scope=openid&state=12345&nonce=12345), but there's no detail. Then I came across [this page](https://agramont.net/blog/devops-intro-deploy-net-aspire-azure-github-actions/) with a wonderful [screenshot](https://agramont.net/_astro/netaspire-azportal02.Bq2W5-R5_Z1otbtp.webp).

![netaspire-azportal02 Bq2W5-R5](https://github.com/user-attachments/assets/4dfc81ad-75dd-4cc8-a6bd-a1b0210e17a0)

Turns out it was deployed all along, I just didn't know where to look.
